RecoHGCal/TICL/interface/PatternRecognitionAlgoBase.h

```diff
@@ -26,14 +26,28 @@ namespace ticl {
         : algo_verbosity_(conf.getParameter<int>("algo_verbosity")) {}
     virtual ~PatternRecognitionAlgoBase(){};
 
-    virtual void makeTracksters(const edm::Event& ev,
-                                const edm::EventSetup& es,
-                                const std::vector<reco::CaloCluster>& layerClusters,
-                                const std::vector<float>& mask,
-                                const edm::ValueMap<float>& layerClustersTime,
-                                const TICLLayerTiles& tiles,
-                                const std::vector<TICLSeedingRegion>& regions,
-                                std::vector<Trackster>& result) = 0;
+    struct Inputs {
+      const edm::Event& ev;
+      const edm::EventSetup& es;
+      const std::vector<reco::CaloCluster>& layerClusters;
+      const std::vector<float>& mask;
+      const edm::ValueMap<float>& layerClustersTime;
+      const TICLLayerTiles& tiles;
+      const std::vector<TICLSeedingRegion>& regions;
+      std::vector<Trackster>& result;
+
+      Inputs(const edm::Event& eV,
+             const edm::EventSetup& eS,
+             const std::vector<reco::CaloCluster>& lC,
+             const std::vector<float>& mS,
+             const edm::ValueMap<float>& lT,
+             const TICLLayerTiles& tL,
+             const std::vector<TICLSeedingRegion>& rG,
+             std::vector<Trackster>& rS)
+          : ev(eV), es(eS), layerClusters(lC), mask(mS), layerClustersTime(lT), tiles(tL), regions(rG), result(rS) {}
+    };
+
+    virtual void makeTracksters(const Inputs& input) = 0;
```

Comment:
(since we are touching the interface anyways)
is there a reason to keep ```makeTracksters``` non-const? If some thread-safety can be imposed by the API, it should be worthwhile.